okio/src/commonTest/kotlin/okio/CommonBufferTest.kt

```diff
@@ -0,0 +1,442 @@
+/*
+ * Copyright (C) 2014 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio
+
+import okio.ByteString.Companion.decodeHex
+import kotlin.random.Random
+import kotlin.test.Ignore
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+/**
+ * Tests solely for the behavior of Buffer's implementation. For generic BufferedSink or
+ * BufferedSource behavior use BufferedSinkTest or BufferedSourceTest, respectively.
+ */
+class CommonBufferTest {
+  @Test fun readAndWriteUtf8() {
+    val buffer = Buffer()
+    buffer.writeUtf8("ab")
+    assertEquals(2, buffer.size)
+    buffer.writeUtf8("cdef")
+    assertEquals(6, buffer.size)
+    assertEquals("abcd", buffer.readUtf8(4))
+    assertEquals(2, buffer.size)
+    assertEquals("ef", buffer.readUtf8(2))
+    assertEquals(0, buffer.size)
+    assertFailsWith<EOFException> {
+      buffer.readUtf8(1)
+    }
+  }
+
+  /** Buffer's toString is the same as ByteString's.  */
+  @Ignore @Test fun bufferToString() {
```

Comment:
Buffer toString requires SegmentedByteString which hasn't moved to Kotlin/Common yet. I moved this unit test but marked it as ignored until SegmentedByteString is moved.

Reply:
Can you leave this comment in the code so it's clear what the blocker is